doc/nrf/gs_modifying.rst

```diff
@@ -84,55 +93,127 @@ The following CMake commands can be managed by SES, if they target the ``app`` l
+Hardware-related configuration
+------------------------------
 
-If you work with SES, the :file:`prj.conf` file is read when you open a project.
-This means that after you edit this file, you must re-open your project.
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: Devicetree
+   :end-before: The preprocessed devicetree sources
 
-.. note::
-   It is possible to change the default configuration for a library by changing the :file:`Kconfig` file of the library.
-   However, best practice is to override the configuration in the application configuration file :file:`prj.conf`.
+The preprocessed devicetree sources (stored in :file:`*.dts.pre.tmp`) are parsed by :file:`zephyr/scripts/dts/gen_defines.py` to generate a :file:`devicetree_unfixed.h` header with preprocessor macros.
+
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: header with preprocessor macros.
+   :end-before: This file is just for reference.
+
+The :file:`zephyr.dts` file contains the entire hardware-related configuration of the system in the devicetree format.
+The header file contains the same kind of information, but with defines usable by source code.
+
+For more information, see Zephyr's :ref:`zephyr:dt-guide`.
+
+.. _configure_application_sw:
+
+Software-related configuration
+------------------------------
+
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: Kconfig
+   :end-before: Information from devicetree is available to Kconfig,
+
+Information from devicetree is available to Kconfig, through the functions defined in :file:`zephyr/scripts/kconfig/kconfigfunctions.py`.
+
+The single :file:`.config` file in the :file:`<build_dir>/zephyr/` directory describes the entire software configuration of the constructed binary.
+
+For more information, see Zephyr's :ref:`zephyr:application-kconfig`.
+
+Memory layout configuration
+---------------------------
+
+The Partition Manager is specific to the |NCS|.
+If enabled, it provides the memory layout configuration.
+The layout is impacted by various elements, such as Kconfig configuration options or the presence of child images.
+Partition Manager ensures that all required partitions are in the correct place and have the correct size.
+
+If enabled, the memory layout can be controlled in the following ways:
+
+* Dynamically (default) - In this scenario, the layout is impacted by various elements, such as Kconfig configuration options or the presence of child images.
+  Partition Manager ensures that all required partitions are in the correct place and have the correct size.
+* Statically - In this scenario, you need to provide the static configuration.
+  See :ref:`ug_pm_static` for information about how to do this.
+
+Dynamically generated configurations are supported by default, but you can also specify :ref:`ug_pm_static`.
```

Comment:
I believe this line can be remove now, because you have just listed both options in the bullet list above, and have noted in the bullet list that Dynamically is default.

```suggestion
```